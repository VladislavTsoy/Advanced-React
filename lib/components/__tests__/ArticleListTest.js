import React from 'react';
import ArticleList from '../ArticleList';

import renderer from 'react-test-renderer';

const testProps = {
    articles: {
        a: { 
            id: 'a', 
            title: 'a',
            date: 'a',
            authorId: 'a',
            body: 'a'
        },
        b: { 
            id: 'b', 
            title: 'b',
            date: 'b',
            authorId: 'b',
            body: 'b' 
        }
    },
    store: {
        lookupAuthor: jest.fn(() => ({})),
    },
};

describe('ArticleList', () => {
    it('ArticleList Component Renders Correctly', () => {
        const tree = renderer.create(
            <ArticleList 
                {...testProps}
            />
        ).toJSON();
       
        expect(tree.children.length).toBe(2); // confirms 2 objects were mapped
        expect(tree).toMatchSnapshot();
    });
});