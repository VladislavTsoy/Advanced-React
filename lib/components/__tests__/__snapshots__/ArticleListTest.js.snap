// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ArticleList ArticleList Component Renders Correctly 1`] = `
<div>
  <div
    style={
      Object {
        "borderBottomColor": "#aaa",
        "borderBottomStyle": "solid",
        "borderBottomWidth": 1,
        "marginBottom": 10,
        "paddingBottom": 10,
      }
    }
  >
    <div
      style={
        Object {
          "fontWeight": "bold",
        }
      }
    >
      a
    </div>
    <div
      style={
        Object {
          "color": "#888",
          "fontSize": "0.85em",
        }
      }
    >
      Invalid Date
    </div>
    <div
      style={
        Object {
          "paddingBottom": 10,
          "paddingTop": 10,
        }
      }
    >
      <a>
         
      </a>
    </div>
    <div
      style={
        Object {
          "paddingLeft": 10,
        }
      }
    >
      a
    </div>
  </div>
  <div
    style={
      Object {
        "borderBottomColor": "#aaa",
        "borderBottomStyle": "solid",
        "borderBottomWidth": 1,
        "marginBottom": 10,
        "paddingBottom": 10,
      }
    }
  >
    <div
      style={
        Object {
          "fontWeight": "bold",
        }
      }
    >
      b
    </div>
    <div
      style={
        Object {
          "color": "#888",
          "fontSize": "0.85em",
        }
      }
    >
      Invalid Date
    </div>
    <div
      style={
        Object {
          "paddingBottom": 10,
          "paddingTop": 10,
        }
      }
    >
      <a>
         
      </a>
    </div>
    <div
      style={
        Object {
          "paddingLeft": 10,
        }
      }
    >
      b
    </div>
  </div>
</div>
`;
